src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,77 @@
+package duke.task;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+
+/**
+ * The Deadline task which represents a deadline.
+ *
+ * @author Leong Jia Hao Daniel
+ */
+public class Deadline extends Task {
+
+    protected LocalDate date;
+    protected String time;
+    protected String dateTime;
+
+    /**
+     * The constructor for a Deadline task.
+     *
+     * @param description The description of the task.
+     * @param dateTime The data and time of the task in the d/MM/yyyy format.
+     */
+    public Deadline(String description, String dateTime) {
+        super(description);
+        String[] details = dateTime.split(" ");
+        DateTimeFormatter inputFormat = DateTimeFormatter.ofPattern("d/MM/yyyy");
+        this.date = LocalDate.parse(details[0], inputFormat);
+        this.time = details[1];
```

Comment:
I think the use of `this` here is unnecessary since it is not shadowed by a constructor param. [(Reference: Avoid unnecessary use of this with fields)](https://se-education.org/guides/conventions/java/index.html#variables)

```suggestion
        date = LocalDate.parse(details[0], inputFormat);
        time = details[1];
```